rollup.config.js

```diff
@@ -0,0 +1,93 @@
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import { getPackages } from '@lerna/project'
+import filterPackages from '@lerna/filter-packages'
+import batchPackages from '@lerna/batch-packages'
+import typescript from '@wessberg/rollup-plugin-ts'
+import commonjs from '@rollup/plugin-commonjs'
+
+import resolve from '@rollup/plugin-node-resolve'
+import json from '@rollup/plugin-json'
+import path from 'path'
+import minimist from 'minimist'
+
+const external = [
+  // Threads
+  '@textile/threads-client',
+  '@textile/context',
+  '@textile/threads-id',
+  '@textile/multiaddr',
+  '@textile/security',
+  '@textile/threads',
+  '@textile/transport',
+  // Textile
+  '@textile/buckets',
+  '@textile/crypto',
+  '@textile/grpc-authentication',
+  '@textile/grpc-connection',
+  '@textile/hub',
+  '@textile/hub-filecoin',
+  '@textile/hub-threads-users',
+  '@textile/users',
+  // Others (add externals and valid es modules here)
+  'stream',
+]
+
+/**
+ * @param {string[]}[scope] - packages to only build (if you don't
+ *    want to build everything)
+ * @param {string[]}[ignore] - packages to not build
+ *
+ * @returns {Promise<any[]>} - sorted list of Package objects that
+ *    represent packages to be built.
+ */
+async function getSortedPackages(scope, ignore) {
+  const packages = await getPackages(__dirname)
+  const filtered = filterPackages(packages, scope, ignore, false, true)
+
+  return batchPackages(filtered).reduce((arr, batch) => arr.concat(batch), [])
+}
+
+/**
+ * @returns Promise<RollupConfig>
+ */
+async function main() {
```

Comment:
this is the main config setup. we can add other build types of iife and umd etc.